app/src/main/java/nerd/tuxmobil/fahrplan/congress/serialization/ScheduleChanges.kt

```diff
@@ -0,0 +1,149 @@
+package nerd.tuxmobil.fahrplan.congress.serialization
+
+import info.metadude.android.eventfahrplan.network.models.Session as SessionNetworkModel
+
+object ScheduleChanges {
+
+    /**
+     * Returns a new list of sessions. Each session is flagged as
+     * ["new"][SessionNetworkModel.changedIsNew], ["canceled"][SessionNetworkModel.changedIsCanceled]
+     * or according to the changes detected when comparing it to its equivalent from the [oldSessions] list.
+     * If any session change is detected then the [onChangesFound] is invoked.
+     */
+    fun List<SessionNetworkModel>.toFlaggedSessions(oldSessions: List<SessionNetworkModel>, onChangesFound: () -> Unit): List<SessionNetworkModel> {
```

Comment:
I am not totally happy with the method name, too. I haven't considered using `with` in a method name yet because it is reserved in Kotlin. But here it does sound good to me.

I experimented with returning a `Pair<List<Session>, Boolean` but then like the current method interface better. Not sure if I like the extra data class better.

Reply:
Well, a `Pair` works, too. It's the lazy way and much less descriptive. But then you only have two values with distinct types. So it's not too much work to figure out what's what when reading the code.

Passing a lambda that is conditionally executed is imho not great API design. You also found out it's not great for testing, which is why you basically wrapped the API for your tests to get an API that is more or less what I'm suggesting, too.

Reply:
:heavy_check_mark: Changed in 8939692fa05a4edbea751479d2cfac82110af40e.